rmw_opendds_cpp/src/rmw_service.cpp

```diff
@@ -97,30 +98,24 @@ rmw_create_service(
   const char * service_name,
   const rmw_qos_profile_t * qos_profile)
 {
-  RMW_CHECK_FOR_NULL_WITH_MSG(node, "node is null", return nullptr);
-  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(node, node->implementation_identifier, opendds_identifier, return nullptr)
-
-  auto node_info = static_cast<OpenDDSNodeInfo *>(node->data);
-  RMW_CHECK_FOR_NULL_WITH_MSG(node_info, "node_info is null", return nullptr);
-  RMW_CHECK_FOR_NULL_WITH_MSG(node_info->participant.in(), "participant is null", return nullptr);
-  RMW_CHECK_FOR_NULL_WITH_MSG(node_info->publisher_listener, "publisher_listener is null", return nullptr);
-  RMW_CHECK_FOR_NULL_WITH_MSG(node_info->subscriber_listener, "subscriber_listener is null", return nullptr);
-
+  auto dds_node = OpenDDSNode::get_from(node);
+  if (!dds_node) {
+    return nullptr; // error set in get_from
```

Comment:
print error? I think most people would consider returning nullptr as an error but it could also be considered normal behavior. I think in this case since it's an error we want to print at this level.

Reply:
It's done in get_from() and the comment indicates.